src/ext/java/org/opentripplanner/ext/siri/updater/SiriETGooglePubsubUpdater.java

```diff
@@ -339,9 +340,14 @@ public void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
                             getTimeSinceStartupString());
                 }
 
-                saveResultOnGraph.execute(graph -> {
-                    snapshotSource.applyEstimatedTimetable(graph, feedId, false, estimatedTimetableDeliveries);
-                });
+                try {
+                    saveResultOnGraph.execute(graph -> {
+                        snapshotSource.applyEstimatedTimetable(graph, feedId, false, estimatedTimetableDeliveries);
+                    }).get();
```

Comment:
I am a bit concerned for your performance because of this.

the future.get() is a blocking call (Which i guess is the point here) however i wonder if this will create a lot of blocking threads.

I wonder if refactoring this to use a CompletableFuture in order to have the ack method as a callback on completion.


Our implementation does not need this blocking call - and if I understand correctly, our implementation will not be affected so I wont block the PR because of this.

Reply:
The problem is that we need the blocking behaviour in order to not run the a loop continuously here https://github.com/opentripplanner/OpenTripPlanner/blob/c799590f95975ee44b2f34e498418a8150abd82e/src/ext/java/org/opentripplanner/ext/siri/updater/SiriETGooglePubsubUpdater.java#L203-L207

Reply:
I understand the problem statement - we also block on our initialization statement - We just dont block on all the subsequent messages. This is however the consequence of this change.

I will approve this as you are OK with this compromise.